torch/distributions/kl.py

```diff
@@ -210,6 +211,24 @@ def _kl_exponential_exponential(p, q):
     return t1 + rate_ratio - 1
 
 
+@register_kl(ExponentialFamily, ExponentialFamily)
+def _kl_expfamily_expfamily(p, q):
+    if not type(p) == type(q):
+        raise NotImplementedError("The cross KL-divergence between different exponential families cannot \
+                            be computed")
+    p_nparams = [Variable(np.data, requires_grad=True) for np in p._natural_params]
+    q_nparams = q._natural_params
+    lg_normal = p._log_normalizer(*p_nparams)
+    gradients = torch.autograd.grad(lg_normal.sum(), p_nparams, create_graph=True)
+    result = q._log_normalizer(*q_nparams) - lg_normal.clone()
+    for pnp, qnp, g in zip(p_nparams, q_nparams, gradients):
+        term = (qnp - pnp) * g
+        for _ in range(len(q.event_shape)):
+            term = term.sum(-1)
```

Comment:
@fritzo added a helper to compute this sum more efficiently in another PR. I'll merge that one first, and then please rebase and update this code

Reply:
Sure. Thank you.